src/Data/Aeson/Types/FromJSON.hs

```diff
@@ -251,7 +251,10 @@ class GFromJSON arity f where
     -- or 'liftParseJSON' (if the @arity@ is 'One').
     gParseJSON :: Options -> FromArgs arity a -> Value -> Parser (f a)
 
+class GOmittedDefault arity f where
```

Comment:
I probably was unclear: Don't add `gOmittedDefault` nor `gOmitField`. The generic deriving is cumbersome already. Unwrapping single constructors isn't a huge task to do manually, and I argue it's better to always be explicit which value will be omitted.